05/app.js

```diff
@@ -0,0 +1,45 @@
+function Student(initFistName, initLastName) {
+    this.firstName = initFistName;
+    this.lastName = initLastName;
+    this.grades = {};
+};
+
+Student.prototype.addGrade = function (subject, grade) {
+    if (typeof this.grades[subject] === 'undefined') {
+        this.grades[subject] = [];
+    }
+    this.grades[subject].push(grade);
+};
+
+Student.prototype.getAverageGrade = function (subject) {
+    let sum = 0;
+    let count = 0;
+    if (subject) {
+        this.grades[subject].forEach(function (num) {
+            sum += num;
+            count++;
+        });
```

Comment:
Zwróć uwagę, że ta część powtarza się z tą z linii 26-29. W takiej sytuacji warto zrobić dodatkową metodę i wywoływać w obu miejscach. To spowoduje, że zmiana "algorytmu" liczenia będzie wymagać zmiany tylko w jednym miejscu kodu programu. Zawsze staramy się do tego dążyć :) Wprowadzisz zmianę?